tedge_config/src/tedge_config.rs

```diff
@@ -14,13 +15,13 @@ pub struct TEdgeConfig {
 
 impl ConfigSettingAccessor<DeviceIdSetting> for TEdgeConfig {
     fn query(&self, _setting: DeviceIdSetting) -> ConfigSettingResult<String> {
-        self.data
-            .device
-            .id
-            .clone()
-            .ok_or(ConfigSettingError::ConfigNotSet {
-                key: DeviceIdSetting::KEY,
-            })
+        let cert_path = self.query(DeviceCertPathSetting)?;
+        let pem = PemCertificate::from_pem_file(cert_path)
+            .map_err(|err| cert_error_into_config_error(DeviceIdSetting::KEY, err))?;
+        let device_id = pem
+            .subject_common_name()
+            .map_err(|err| cert_error_into_config_error(DeviceIdSetting::KEY, err))?;
+        Ok(device_id)
     }
 
     fn update(&mut self, _setting: DeviceIdSetting, _value: String) -> ConfigSettingResult<()> {
```

Comment:
The error messages of update and unset are no longer correct because you can also use a certificate created not by tedge cert.
There should be better phrasing for sure, this is my example.
`tedge config` cannot set the device.id. The device ID is derived from the certificate.

Reply:
Good point!

What about:

```
The device id is read from the device id and cannot be set directly.
To set 'device.id' to some <id>, you can use `tedge cert create --device-id <id>`.
```

Reply:
I like your proposal!